### ISSUE TITLE
Sphinx builds fails at pickling environment due to pickling error at pickle.dump(self.env, f, pickle.HIGHEST_PROTOCOL) with no useful error message

### ISSUE THREAD

**Issue by DanielPBak** [sphinx-doc/sphinx#5958]
```
C:\Users\danie01.AD\PycharmProjects\idmlib\doc (fox-45481 -> origin)
(inst1) λ "C:\Program Files\Python 3.5\Scripts\sphinx-build.exe" -b html -d build/doctrees  source build/html
Running Sphinx v1.8.3
WARNING: Unknown configure key: latex_elements['MyPath']. ignored.
WARNING: Unknown configure key: latex_elements['TitleLineOne']. ignored.
WARNING: Unknown configure key: latex_elements['draft']. ignored.
WARNING: Unknown configure key: latex_elements['TitleLineTwo']. ignored.
loading intersphinx inventory from https://docs.python.org/3.5/objects.inv...
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 69 source files that are out of date
updating environment: 69 added, 0 changed, 0 removed
reading sources... [100%] wfobject/resource/index
C:\Program Files\Hitachi ID\IDM Suite\inst1\idmlib\agent.py:docstring of idmlib.agent.Agent:17: WARNING: Unexpected indentation.
C:\Program Files\Hitachi ID\IDM Suite\inst1\idmlib\agent.py:docstring of idmlib.agent.Agent:29: WARNING: Block quote ends without a blank line; unexpected unindent.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.PasswordStrengthCheck:3: WARNING: Definition list ends without a blank line; unexpected unindent.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.PasswordStrengthCheck:5: WARNING: Unexpected indentation.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.PasswordStrengthCheck_SC:3: WARNING: Definition list ends without a blank line; unexpected unindent.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.PasswordStrengthCheck_SC:5: WARNING: Unexpected indentation.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.PasswordStrengthTest:7: WARNING: Unexpected indentation.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.PasswordStrengthTest_SC:7: WARNING: Unexpected indentation.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleCreate:7: WARNING: Unexpected indentation.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleCreate:8: WARNING: Block quote ends without a blank line; unexpected unindent.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleCreate:10: WARNING: Unexpected indentation.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleCreate:12: WARNING: Block quote ends without a blank line; unexpected unindent.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleCreate_SC:7: WARNING: Unexpected indentation.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleCreate_SC:8: WARNING: Block quote ends without a blank line; unexpected unindent.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleCreate_SC:10: WARNING: Unexpected indentation.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleCreate_SC:12: WARNING: Block quote ends without a blank line; unexpected unindent.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleList:5: WARNING: Unexpected indentation.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleList:6: WARNING: Block quote ends without a blank line; unexpected unindent.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleList_SC:5: WARNING: Unexpected indentation.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleList_SC:6: WARNING: Block quote ends without a blank line; unexpected unindent.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleResourceAdd:8: WARNING: Unexpected indentation.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleResourceAdd:10: WARNING: Block quote ends without a blank line; unexpected unindent.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleResourceAdd:21: WARNING: Definition list ends without a blank line; unexpected unindent.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleResourceAdd_SC:8: WARNING: Unexpected indentation.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleResourceAdd_SC:10: WARNING: Block quote ends without a blank line; unexpected unindent.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleResourceAdd_SC:21: WARNING: Definition list ends without a blank line; unexpected unindent.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleUpdate:5: WARNING: Unexpected indentation.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleUpdate:6: WARNING: Block quote ends without a blank line; unexpected unindent.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleUpdate:8: WARNING: Unexpected indentation.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleUpdate:10: WARNING: Block quote ends without a blank line; unexpected unindent.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleUpdate_SC:5: WARNING: Unexpected indentation.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleUpdate_SC:6: WARNING: Block quote ends without a blank line; unexpected unindent.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleUpdate_SC:8: WARNING: Unexpected indentation.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.RoleUpdate_SC:10: WARNING: Block quote ends without a blank line; unexpected unindent.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.TargetAttributeGet:3: WARNING: Definition list ends without a blank line; unexpected unindent.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.TargetAttributeGet:6: WARNING: Unexpected indentation.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.TargetAttributeGet:11: WARNING: Block quote ends without a blank line; unexpected unindent.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.TargetAttributeGet_SC:3: WARNING: Definition list ends without a blank line; unexpected unindent.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.TargetAttributeGet_SC:6: WARNING: Unexpected indentation.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.TargetAttributeGet_SC:11: WARNING: Block quote ends without a blank line; unexpected unindent.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.UserAnswersValidate:9: WARNING: Unexpected indentation.
C:\Program Files\Hitachi ID\IDM Suite\inst1\lib\idapi_h.py:docstring of lib.idapi_h.APIBase.UserAnswersValidate_SC:9: WARNING: Unexpected indentation.
C:\Program Files\Hitachi ID\IDM Suite\inst1\idmlib\idmobject.py:docstring of idmlib.idmobject.Request.api_submit:9: WARNING: Field list ends without a blank line; unexpected unindent.
looking for now-outdated files... none found
pickling environment...
Exception occurred:
  File "c:\program files\python 3.5\lib\site-packages\sphinx\builders\__init__.py", line 377, in build
    pickle.dump(self.env, f, pickle.HIGHEST_PROTOCOL)
_pickle.PicklingError: Can't pickle <class 'typing.ChangeSet'>: attribute lookup ChangeSet on typing failed
The full traceback has been saved in C:\Users\danie01.AD\AppData\Local\Temp\sphinx-err-n5g09vdh.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!

Build finished. The HTML pages are in build/html.
```

And then build/html is empty.

I am on Sphinx 1.8.3, and it looks like ChangeSet is defined inside Sphinx, so I'm curious about what in my configuration could be causing Sphinx to try to pickle something internal which is not picklable. Does 1.8.3 not support Python 3.5?

**Comment by DanielPBak** [comment]
Changing from Python 3.5.0 to Python 3.5.4 fixed this issue.

**Comment by tk0miya** [comment]
OMG, it's a bug of Python-3.5.0. We should not use `typing.NamedTuple` ...
https://bugs.python.org/issue25665

**Comment by tk0miya** [comment]
Fixed by #5978 
Thank you for reporting.
